### PR TITLE
test: Avoid racily injecting code in dashboard tests

### PR DESCRIPTION
**Pull request by stefwalter**
We should avoid injecting code, which may happen during a page
load during the dashboard tests.

### CODE REVIEW

**Comment by stefwalter** [comment]
Trying to solve this problem:

```
# ----------------------------------------------------------------------
# testBasic (check_dashboard.TestBasicDashboard)
#
ERROR: fatal problem: Network access is disabled.
Resource Error: Network access is disabled. http://10.111.115.125:9090/cockpit/$be55352332739898ce8a0bd27a71c56db3cc001c/shell/images/server-error.png
Page error: ReferenceError: Can't find variable: dashboard_addresses
undefined 2 
 3 
 3 
```

**Comment by martinpitt** [comment]
This collides with PR #5963, so we need to adjust one after landing the other.

**Comment by mvollmer** [comment]
Now we see: ```Page error: ReferenceError: Can't find variable: $```

**Comment by mvollmer** [comment]
@stefwalter, the test fails with `Error: PhantomJS or driver broken`.  Shouldn't that lead to a retry?

**Comment by mvollmer** [comment]
I think this doesn't fundamentally solve the issue: the page hasn't properly been loaded, and the test can't pass.  We can just tune the kind of failure we get, no?  If the retry on "Network access disabled", the fix here is not necessary, I think.

**Comment by mvollmer** [comment]
> the test fails with Error: PhantomJS or driver broken. Shouldn't that lead to a retry?

It does retry, but this is not visible from the prettified log.html, for some reason.

**Comment by stefwalter** [comment]
> It does retry, but this is not visible from the prettified log.html, for some reason.

Indeed. That should be fixed in either in ```log.html``` or in the ```TestResult.startTest```

File: test/verify/check-dashboard [hunk 1]
```diff
@@ -48,7 +40,8 @@ class DashBoardHelpers:
     def wait_dashboard_addresses(self, b, expected):
         b.wait_js_func(
             """(function (expected) {
-            return expected.sort().toString() == dashboard_addresses().sort().toString();
+            var addresses = $('#dashboard-hosts .list-group-item').map(function(i,e) { return $(e).data("address"); }).get();
```
Comment (dperpeet) [inline test/verify/check-dashboard]: nitpick: formatting - I can post a fixup commit

**Comment by dperpeet** [comment]
I suggest this fixup: https://github.com/dperpeet/cockpit/commit/15383c6939be8f5c2d344b0cfbe091b6d5826377

**Comment by mvollmer** [comment]
>[...] the fix here is not necessary, I think.

But it's still an improvement.

**Comment by stefwalter** [comment]
Included @dperpeet's changes and removed use of jQuery.

**Comment by martinpitt** [comment]
Remaining two test failures were in Network, unrelated to this change. Merging.

**Pull request closed without merging**